service/lambda/api_op_AddLayerVersionPermission.go

```diff
@@ -208,6 +209,52 @@ func (m *opAddLayerVersionPermissionResolveEndpointMiddleware) HandleSerialize(c
 		)
 	}
 
+	authSchemes, err := internalauth.GetAuthenticationSchemes(&resolvedEndpoint.Properties)
+	if err != nil || len(authSchemes) == 0 {
```

Comment:
**fix**: The rules are not required to contain auth schemes, the logic here should just update the context as needed _if_ there is an auth scheme we support